From 61e57b19346bde55925bcc91ace2b09a362bcb1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Sat, 16 Apr 2022 00:44:03 +0200
Subject: [PATCH 48/52] smb2: remove unneeded compat code

thread_local is unavailable to 32bit on iOS and the code in question is
not needed as we have the latest libsmb2, so let's remove it.
---
 modules/access/smb2.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 7373e5904a..66df992cfb 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -190,34 +190,6 @@ smb2_set_error(struct vlc_smb2_op *op, const char *psz_func, int err)
 
 #define VLC_SMB2_STATUS_DENIED(x) (x == -ECONNREFUSED || x == -EACCES)
 
-#if defined (__ELF__) || defined (__MACH__) /* weak support */
-/* There is no way to know if libsmb2 has these new symbols and we don't want
- * to increase the version requirement on VLC 3.0, therefore implement a weak
- * compat version. */
-const t_socket *
-smb2_get_fds(struct smb2_context *smb2, size_t *fd_count, int *timeout);
-int
-smb2_service_fd(struct smb2_context *smb2, int fd, int revents);
-
-__attribute__((weak)) const t_socket *
-smb2_get_fds(struct smb2_context *smb2, size_t *fd_count, int *timeout)
-{
-    (void) timeout;
-    static thread_local t_socket fd;
-
-    *fd_count = 1;
-    fd = smb2_get_fd(smb2);
-    return &fd;
-}
-
-__attribute__((weak)) int
-smb2_service_fd(struct smb2_context *smb2, int fd, int revents)
-{
-    (void) fd;
-    return smb2_service(smb2, revents);
-}
-#endif
-
 static int
 vlc_smb2_mainloop(struct vlc_smb2_op *op)
 {
-- 
2.32.0 (Apple Git-132)

